base/linalg/exceptions.jl

```diff
@@ -29,10 +29,29 @@ function Base.showerror(io::IO, ex::ARPACKException)
     end
 end
 
+"""
+   SingularException(info)
+
+The matrix passed is a singular matrix. Optional argument INFO is an INTEGER
+= 0:  successful exit
+< 0:  if INFO = -i, the i-th argument had an illegal value
+> 0:  if INFO = i, U(i,i) is exactly zero. The factorization
+      has been completed, but the factor U is exactly
+      singular, and division by zero will occur if it is used
+      to solve a system of equations.
+"""
 struct SingularException <: Exception
     info::BlasInt
 end
 
+function Base.showerror(io::IO, ex::SingularException)
+    print(io, "SingularException: ")
+    if ex.info > 0
+        print(io,"matrix is singular")
+    elseif ex.info <0
+        print(io,"argument $(ex.info) had illegal value")
```

Comment:
This will never happen; we throw a LAPACK exception before we reach this.

Reply:
Might as well test for it anyway – defensive programming and all that.

Reply:
That test should go into the constructor.

Reply:
If it will always be >0 so I think there is no need for checking at all?

Reply:
I think we should leave it out here. We check all values *before* calling LAPACK so we should not even get negative values back from LAPACK and as mentioned by @fredrikekre, we check the LAPACK return values so this is already asserted.